src/Storefront/Controller/CookieController.php

```diff
@@ -29,173 +27,92 @@ class CookieController extends StorefrontController
+        }
 
         return $cookieGroups;
     }
 
     /**
-     * @param array<string|int, mixed> $cookieGroups
+     * Transforms the cookie group collection to a format suitable for Twig.
+     *
+     * Ensures that all snippet names and descriptions are initialized or set to empty strings.
      *
      * @return array<string|int, mixed>
      */
-    private function filterGoogleAnalyticsCookie(SalesChannelContext $context, array $cookieGroups): array
+    private function transformCookieGroupForTwig(CookieGroupCollection $cookieGroupCollection): array
     {
-        $salesChannel = $context->getSalesChannel();
-
-        if ($salesChannel->getAnalytics() === null && $salesChannel->getAnalyticsId() !== null) {
-            $criteria = new Criteria([$salesChannel->getAnalyticsId()]);
-            $criteria->setTitle('cookie-controller::load-analytics');
-
-            $salesChannel->setAnalytics(
-                $this->salesChannelAnalyticsRepository->search($criteria, $context->getContext())->getEntities()->first()
-            );
-        }
-
-        if ($salesChannel->getAnalytics()?->isActive() === true) {
-            return $cookieGroups;
-        }
-
-        $filteredGroups = [];
-        foreach ($cookieGroups as $cookieGroup) {
-            if ($cookieGroup['snippet_name'] === 'cookie.groupStatistical') {
-                $cookieGroup = $this->filterCookieGroup('cookie.groupStatisticalGoogleAnalytics', $cookieGroup);
-                if ($cookieGroup !== null) {
-                    $filteredGroups[] = $cookieGroup;
-                }
-
-                continue;
-            } elseif ($cookieGroup['snippet_name'] === 'cookie.groupMarketing') {
-                $cookieGroup = $this->filterCookieGroup('cookie.groupMarketingAdConsent', $cookieGroup);
-                if ($cookieGroup !== null) {
-                    $filteredGroups[] = $cookieGroup;
-                }
-
-                continue;
+        foreach ($cookieGroupCollection as $group) {
+            $this->setDefaultValuesForCookieStruct($group);
+            foreach ($group->entries as $cookieStruct) {
+                $this->setDefaultValuesForCookieStruct($cookieStruct);
             }
+        }
 
-            $filteredGroups[] = $cookieGroup;
+        $result = [];
+        foreach ($cookieGroupCollection as $group) {
+            $result[] = $group->jsonSerialize();
         }
 
-        return $filteredGroups;
+        return $result;
     }
 
-    /**
-     * @param array<string|int, mixed> $cookieGroups
-     *
-     * @return array<string|int, mixed>
-     */
-    private function filterWishlistCookie(string $salesChannelId, array $cookieGroups): array
+    private function setDefaultValuesForCookieStruct(CookieEntry|CookieGroup $cookieStruct): void
     {
-        if ($this->systemConfigService->getBool('core.cart.wishlistEnabled', $salesChannelId)) {
-            return $cookieGroups;
+        if (!isset($cookieStruct->snippetName)) {
```

Comment:
that is also something, I don't like about your change. why do we introduce special handling for cookie objects in the controller? 

obviously we do not have those problems with other structs :man_shrugging:

Reply:
uninitialized properties is not a wanted state :grimacing: 

they are public on your structs. if you access them, without being set before the program will crash. you cannot foresee how they will be used :see_no_evil: